client/my-sites/upgrade-nudge/index.jsx

```diff
@@ -56,13 +57,20 @@ export default React.createClass( {
 		this.props.onClick();
 	},
 
-	componentDidMount() {
-		if ( this.props.event || this.props.feature ) {
-			analytics.tracks.recordEvent( 'calypso_upgrade_nudge_impression', {
-				cta_name: this.props.event,
-				cta_feature: this.props.feature
-			} );
+	shouldDisplay( site ) {
+		if ( site && this.props.feature ) {
```

Comment:
I know you're copying over logic here, but I think it's not behaving how we want when the site is not yet loaded (can be either `false` or `undefined`, and I also think it would be more clear if we removed the nesting.
```js
const { feature, jetpack } = this.props;
if ( ! site ) {
  return false;
}
if ( feature && hasFeature( feature, site.siteID ) ) {
  return false;
}
if ( ! feature && ! isFreePlan( site.plan ) ) {
  return false;
}
if ( ! jetpack && site.jetpack || jetpack && ! site.jetpack ) {
  return false;
}
```
It also might be nice to create an additional utility function for `isPaidPlan` that just negates `isFreePlan`... make the conditionals read more clearly IMHO.